libpod/define/container_inspect.go

```diff
@@ -87,6 +90,51 @@ type InspectContainerConfig struct {
 	SdNotifySocket string `json:"sdNotifySocket,omitempty"`
 }
 
+// UnmarshalJSON allow compatibility with podman V4 API
+func (insp *InspectContainerConfig) UnmarshalJSON(data []byte) error {
+	type Alias InspectContainerConfig
+	aux := &struct {
+		Entrypoint interface{} `json:"Entrypoint"`
+		StopSignal interface{} `json:"StopSignal"`
+		*Alias
+	}{
+		Alias: (*Alias)(insp),
+	}
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	switch entrypoint := aux.Entrypoint.(type) {
+	case string:
+		insp.Entrypoint = []string{entrypoint}
```

Comment:
this would need to at least split the string by spaces, just dumping everything into argv0 is wrong. Not that splitting by spaces would be 100% correct either but it should be better.

Reply:
fixed !